packages/editor-ui/src/modules/nodeTypes.ts

```diff
@@ -0,0 +1,87 @@
+import Vue from 'vue';
+import { ActionContext, Module } from 'vuex';
+import type { INodeTypeDescription } from 'n8n-workflow';
+
+import type { IRootState, INodeTypesState } from '../Interface';
+import { DEFAULT_NODETYPE_VERSION } from '@/constants';
+import { getNodeTypes } from '@/api/nodeTypes';
+
+const module: Module<INodeTypesState, IRootState> = {
+	namespaced: true,
+	state: {
+		nodeTypes: {},
+	},
+	getters: {
+		allNodeTypes: (state): INodeTypeDescription[] => {
+			return Object.values(state.nodeTypes);
+		},
+		nodeType: (state) => (nodeType: string, version?: number): INodeTypeDescription | null => {
+			const foundType = Object.values(state.nodeTypes).find(typeData => {
+				const typeVersion = Array.isArray(typeData.version)
+					? typeData.version
+					: [typeData.version];
+
+				return typeData.name === nodeType && typeVersion.includes(
+					version || typeData.defaultVersion || DEFAULT_NODETYPE_VERSION,
+				);
+			});
+
+			if (foundType === undefined) {
+				return null;
+			}
+
+			return foundType;
+		},
+	},
+	mutations: {
+		setNodeTypes(state, nodeTypesArray: INodeTypeDescription[]) {
+			Vue.set(state, 'nodeTypes', toNodeTypesState(nodeTypesArray));
```

Comment:
`state.nodeTypes = ` should work fine here.. only need to use `Vue.set` when you are digging deeper into object `nodeTypes.innerKey` for example to get reactivity

Reply:
Got it, thanks!